src/main/java/org/assertj/assertions/generator/BaseAssertionGenerator.java

```diff
@@ -81,7 +81,17 @@
   // used to infer the class name from the entry point custom template (specially if the template is custom)
   // - [\s]*    : any number of white space character
   // - \b(.*)\b : capture word
-  private static final Pattern CLASS_NAME_PATTERN = Pattern.compile("public class[\\s]+(\\p{javaJavaIdentifierStart}\\p{javaJavaIdentifierPart}*)\\b");
+  /**
+   * This regexp shall match a java class's name inside an user template.
+   * <p>
+   * For this, we use the two character class {@code javaJavaIdentifierStart} and {@code javaJavaIdentifierPart} to match
+   * a valid name.
+   *
+   * @see java.util.regex.Pattern
+   * @see Character#isJavaIdentifierStart
+   * @see Character#isJavaIdentifierPart
+   */
+  private static final Pattern CLASS_NAME_PATTERN = Pattern.compile("public class[\\s]+(?<CLASSNAME>\\p{javaJavaIdentifierStart}\\p{javaJavaIdentifierPart}*)\\b");
```

Comment:
The comment is good but can be a bit better by also describing the different `[\\s]+` and `?<CLASSNAME>`, it might be obvious for you but it's not yet crystal clear to me (although I get most of it).